src/rshim.c

```diff
@@ -2069,43 +2120,40 @@ rshim_backend_t *rshim_find_by_dev(void *dev)
   return NULL;
 }
 
+/*
+ * Must run with bd->mutex held.
+ *
+ * No need to call this function when value is set to 1. Just set bd->drop_mode
+ * to 1.
+ * */
 int rshim_set_drop_mode(rshim_backend_t *bd, int value)
 {
   int old_value;
   int rt = 0;
 
-  pthread_mutex_lock(&bd->mutex);
   old_value = (int)bd->drop_mode;
   value = !!value;
   if (value == old_value) {
-    pthread_mutex_unlock(&bd->mutex);
     return -EALREADY;
   }
 
-  bd->drop_mode = 0;
-  if (bd->enable_device && bd->enable_device(bd, value ? false : true))
-    bd->drop_mode = 1;
-  else
-    bd->drop_mode = value;
+  bd->drop_mode = value;
```

Comment:
This change removed like 2086 and could cause issue. We might need to change it back.

Reply:
This change is the newchange per our latest discussion:

_Liming Sun
7/2/2024 2:32 PM
In the ownership function, we can leave the drop mode as 1, but can call bd->enable_device(true)_

You suggested when we enter drop_mode, we call bd->enable_device(true). But if drop_mode is 0, the device is already enabled, so no need to call bd->enable_device() at all?

Reply:
Not here, it's another place (see comment 'This is the place to enable the device' below).

Reply:
Agree it's less impact that way. Will change.